spec/spec_helper.rb

```diff
@@ -277,6 +312,27 @@ def assert_faster_than(competition, records, &blk)
     upsert_time.should be < ar_time
     $stderr.puts "   Upsert was #{((ar_time - upsert_time) / ar_time * 100).round}% faster than #{competition}"
   end
+
+  def clone_ar_class(klass, table_name)
+    u = Upsert.new $conn, klass.table_name
+    new_table_name = [*table_name].compact
+    # AR's support for quoting of schema and table names is horrendous
+    # schema.table and schema.`table` are considiered different names on MySQL, but
+    # schema.table and schema."table" are correctly considered the same on Postgres
+    new_table_name[-1] = u.connection.quote_ident(new_table_name[-1]) if new_table_name[-1].to_s.index('.')
+    new_table_name = new_table_name.join('.')
+    cls = Class.new(klass)
+    cls.class_eval do
+      self.table_name = new_table_name
```

Comment:
i have never set activerecord's `table_name` to anything other than an unquoted string

is fully qualified schema a supported option?

i just don't know how this would work with (imagine that) `quoted_table_name`... would it detect the quotes? would it parse out the schema?

i would expect

```
# set search_path to "bar"
class Foo < ActiveRecord::Base
  self.table_name = 'foo'
end
```

Now Foo would look in either public.foos or bar.foos

Reply:
Depending on database driver (thanks MySQL) it may or may not do funny things with the string.  Was to get the specs passing until we can revisit the whole thing.